application/src/main/java/org/togetherjava/tjbot/features/basic/Starboard.java

```diff
@@ -0,0 +1,77 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.entities.channel.middleman.GuildChannel;
+import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.StarboardConfig;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.EventReceiver;
+
+import java.util.Optional;
+
+import static org.togetherjava.tjbot.db.generated.tables.StarboardMessages.STARBOARD_MESSAGES;
+
+public class Starboard extends ListenerAdapter implements EventReceiver {
+
+    private static final Logger logger = LoggerFactory.getLogger(Starboard.class);
+    private final StarboardConfig config;
+    private final Database database;
+
+    public Starboard(Config config, Database database) {
+        this.config = config.getStarboard();
+        this.database = database;
+    }
+
+    @Override
+    public void onMessageReactionAdd(@NotNull MessageReactionAddEvent event) {
+        String emojiName = event.getEmoji().asCustom().getName();
+        Guild guild = event.getGuild();
+        if (ignoreMessage(emojiName, guild, event.getGuildChannel())) {
+            return;
+        }
+        Optional<TextChannel> starboardChannel = getStarboardChannel(guild);
+        if (starboardChannel.isEmpty()) {
+            logger.warn("There is no channel for the starboard in the guild with the name {}",
+                    config.getChannelName());
+            return;
+        }
+        if (database.read(context -> context.selectFrom(STARBOARD_MESSAGES)
+            .fetch("message_id")
+            .contains(event.getMessageIdLong()))) {
+            database.write(context -> context.newRecord(STARBOARD_MESSAGES)
+                .setMessageId(event.getMessageIdLong()));
+        }
```

Comment:
whats the point of the database entry? its not used anywhere right now. id guess u want to check whether it exists to not duplicate-post? but that check seems to be missing right now.

also, u will have to additionally put an in-memory cache as layer above the DB to prevent checking the DB a million times in 5 seconds when someone posts an announcement. u can look in some of the other classes for that, search for `Caffeine`(cache).

Reply:
The database is being used for the check, that is what the if is for. However the current approach seems to be insanely slow as I fetch the entire list of entries. 

I'll look into it.

Reply:
i think your missing my point on the question. what is the point of the database entry? it does not control any logic right now.
you have:
```java
if (database doesnt contain it) {
  put it into database
}
```
but thats it. what point does it serve, sitting in the db? theres nothing in ur code that would control anything based on it.

u likely intended to stop transfering the message if its contained in the db, i.e.
```java
if (database does not contain it) {
  copy message to starboard
}
```
but u dont have that. you always copy the message to the starboard, regardless of database

Reply:
Oh oops. Ur right mb 😅